src/ncp/ncp_spinel.cpp

```diff
@@ -443,6 +453,42 @@ otError NcpSpinel::SendEncodedFrame(void)
     return error;
 }
 
+otError NcpSpinel::ParseIp6AddressTable(const uint8_t               *aBuf,
+                                        uint16_t                     aLength,
+                                        std::vector<Ip6AddressInfo> &aAddressTable)
+{
+    otError             error = OT_ERROR_NONE;
+    ot::Spinel::Decoder decoder;
+
+    VerifyOrExit(aBuf != nullptr, error = OT_ERROR_INVALID_ARGS);
+    decoder.Init(aBuf, aLength);
+
+    while (!decoder.IsAllReadInStruct())
+    {
+        Ip6AddressInfo      cur;
+        const otIp6Address *addr;
+        uint8_t             prefixLength;
+        uint32_t            preferredLft;
```

Comment:
nit: any reason to abbreviate? When reading, I wasn't sure if it was "Left" or "Lifetime".
```suggestion
        uint32_t            preferredLifetime;
```